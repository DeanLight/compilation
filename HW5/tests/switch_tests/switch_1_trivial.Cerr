[[[ added func: print with 1 params and label: ]]]
[[[ added func: printi with 1 params and label: ]]]
print param:2(1)
printi param:3(1)
[FIRST_PROGRAM_POINT]
[set_func_label] label print for function print
[set_func_label] label printi for function printi
(END_OF)  [FIRST_PROGRAM_POINT]
Lex Ate token void__
Lex Ate token  id__main__
Lex Ate token  (__
Lex Ate token )__
<<FuncHead_semantic, id:[main]>>
[[[ added func: main with 0 params and label: ]]]
FuncHead_IR for func: main
got label label_29 in line: -1
[set_func_label] label label_29 for function main
[get_func_label] main's label is: label_29
written label: label_29
END_OF [FuncHead_IR] 
Lex Ate token  {__
Lex Ate token switch__
Lex Ate token  (__
Lex Ate token  num__1__
Lex Ate token )__
[Exp_IR] Exp -> Num:1
[get_func_label] main's label is: label_29
{{{{ NOT RELEVANT:label_29}}}}
END_OF [Exp_IR] Exp -> Num
Lex Ate token  {__
[MARKER IR]: label: bp_label_0
Lex Ate token case__
Lex Ate token  num__1__
Lex Ate token  :__
Lex Ate token  id__print__
Lex Ate token  (__
CallHeader_IR
[Emitter] store_registers
END_OF CallHeader_IR
Lex Ate token  str__"one\n"__
[Exp_IR] Exp -> string"one\n"
END_OF [Exp_IR] Exp -> string
Lex Ate token )__
<<ExpList_Semantic>> explist -> exp
exp is of type STRING
[ExpList_IR: ExpList->Exp]: 
<<CallSemantics [ID(Exp)] >>
tmpParams: 1
tmpActP: 1
tmptmp: 1
[Call_IR] : with arguments print
[get_func_label] print's label is: print
found a string typed parameter print
[Emitter] REstore_registers: 1
ERROR: Reg not given yet. req:0 while max used is 0
END_OF [Call_IR]
Lex Ate token  ;__
[Statement_IR Call] nestlistSize: 0
[Statement_next_patcher_IR] previously held nextlist size:0
[Statement_next_patcher_IR] currently held nextlist size:1
[Statement_IR Call] nestlistSize: 1
[Statement_IR: States->SignleState]  nextlistSize=1
Lex Ate token  }__
Lex Ate token  id__print__
