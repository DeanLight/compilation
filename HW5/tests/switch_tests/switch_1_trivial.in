void main(){
	switch(1){
		case 1:
		print("one\n");
	};
	print("two\n");
	switch(3){
		case 1:
		print("one\n");
	};
	switch(3){
		case 3 b:
		print("three\n");
	};
	switch(4 b){
		case 4 :
		print("four");
	};
}