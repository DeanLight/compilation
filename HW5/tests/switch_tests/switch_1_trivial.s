.data
pvufne1:	.asciiz "Error division by zero
"
string_label_0:	.asciiz "one\n"
string_label_1:	.asciiz "one\n"
string_label_2:	.asciiz "two\n"
string_label_3:	.asciiz "two\n"
string_label_4:	.asciiz "one\n"
string_label_5:	.asciiz "one\n"
string_label_6:	.asciiz "three\n"
string_label_7:	.asciiz "three\n"
string_label_8:	.asciiz "four"
string_label_9:	.asciiz "four"
.text
	#first program point
main:
	#initialize fp
	move	$fp,$sp
	jal	label_29
	#exiting...
li $v0, 10
syscall
	#print_func:
print:
lw $a0,0($sp)
li $v0,4
syscall
jr $ra
	#printi func:
printi:
lw $a0,0($sp)
li $v0,1
syscall
jr $ra
	#div by 0 handler:
nayalo:
	li	$v0, 4
	la	$a0, pvufne1
	syscall
	#exiting...
li $v0, 10
syscall
	#Func main:
label_29:
	li	$t0,1
	#switch!
	j	bp_label_1
	#marker Label
bp_label_0:
	#func header store regs before call
	addiu $sp, $sp, -4
	sw $t0,4($sp)
	#stored 1 registers
	#caught a string
	#preparing to call print
	addiu $sp, $sp, -4
	sw $fp, ($sp)
	addiu $sp, $sp, -4
	sw $ra, ($sp)
	#loading str address to stack
	addiu $sp, $sp, -4
	la $v0, string_label_1
	sw $v0, ($sp)
	move	$fp,$sp
	#jumping to func: print
	jal	print
	#returned from func: print
	#poping 1 params from stack 
	addiu $sp, $sp, 4
	#restoring ra and fp
	lw $ra, ($sp)
	addiu $sp, $sp, 4
	lw $fp, ($sp)
	addiu $sp, $sp, 4
	#restoring 1 previously used registers
	lw $t0,4($sp)
	#Moving funcRes (if exists) to next free register
	#finished calling print
	#end of statement jump
	j	bp_label_2
	#switch init label
bp_label_1:
	beq $t0,1,bp_label_0
bp_label_2:
	#end of statement jump
	j	bp_label_3
	#marker Label
bp_label_3:
	#func header store regs before call
	addiu $sp, $sp, -4
	sw $t0,4($sp)
	#stored 1 registers
	#caught a string
	#preparing to call print
	addiu $sp, $sp, -4
	sw $fp, ($sp)
	addiu $sp, $sp, -4
	sw $ra, ($sp)
	#loading str address to stack
	addiu $sp, $sp, -4
	la $v0, string_label_3
	sw $v0, ($sp)
	move	$fp,$sp
	#jumping to func: print
	jal	print
	#returned from func: print
	#poping 1 params from stack 
	addiu $sp, $sp, 4
	#restoring ra and fp
	lw $ra, ($sp)
	addiu $sp, $sp, 4
	lw $fp, ($sp)
	addiu $sp, $sp, 4
	#restoring 1 previously used registers
	lw $t0,4($sp)
	#Moving funcRes (if exists) to next free register
	#finished calling print
	#end of statement jump
	j	bp_label_4
	#marker Label
bp_label_4:
	li	$t1,3
	#switch!
	j	bp_label_6
	#marker Label
bp_label_5:
	#func header store regs before call
	addiu $sp, $sp, -8
	sw $t0,8($sp)
	sw $t1,4($sp)
	#stored 2 registers
	#caught a string
	#preparing to call print
	addiu $sp, $sp, -4
	sw $fp, ($sp)
	addiu $sp, $sp, -4
	sw $ra, ($sp)
	#loading str address to stack
	addiu $sp, $sp, -4
	la $v0, string_label_5
	sw $v0, ($sp)
	move	$fp,$sp
	#jumping to func: print
	jal	print
	#returned from func: print
	#poping 1 params from stack 
	addiu $sp, $sp, 4
	#restoring ra and fp
	lw $ra, ($sp)
	addiu $sp, $sp, 4
	lw $fp, ($sp)
	addiu $sp, $sp, 4
	#restoring 2 previously used registers
	lw $t0,8($sp)
	lw $t1,4($sp)
	#Moving funcRes (if exists) to next free register
	#finished calling print
	#end of statement jump
	j	bp_label_7
	#switch init label
bp_label_6:
	beq $t1,1,bp_label_5
bp_label_7:
	#end of statement jump
	j	bp_label_8
	#marker Label
bp_label_8:
	li	$t2,3
	#switch!
	j	bp_label_10
	#marker Label
bp_label_9:
	#func header store regs before call
	addiu $sp, $sp, -12
	sw $t0,12($sp)
	sw $t1,8($sp)
	sw $t2,4($sp)
	#stored 3 registers
	#caught a string
	#preparing to call print
	addiu $sp, $sp, -4
	sw $fp, ($sp)
	addiu $sp, $sp, -4
	sw $ra, ($sp)
	#loading str address to stack
	addiu $sp, $sp, -4
	la $v0, string_label_7
	sw $v0, ($sp)
	move	$fp,$sp
	#jumping to func: print
	jal	print
	#returned from func: print
	#poping 1 params from stack 
	addiu $sp, $sp, 4
	#restoring ra and fp
	lw $ra, ($sp)
	addiu $sp, $sp, 4
	lw $fp, ($sp)
	addiu $sp, $sp, 4
	#restoring 3 previously used registers
	lw $t0,12($sp)
	lw $t1,8($sp)
	lw $t2,4($sp)
	#Moving funcRes (if exists) to next free register
	#finished calling print
	#end of statement jump
	j	bp_label_11
	#switch init label
bp_label_10:
	beq $t2,3,bp_label_9
bp_label_11:
	#end of statement jump
	j	bp_label_12
	#marker Label
bp_label_12:
	li	$t3,4
	#switch!
	j	bp_label_14
	#marker Label
bp_label_13:
	#func header store regs before call
	addiu $sp, $sp, -16
	sw $t0,16($sp)
	sw $t1,12($sp)
	sw $t2,8($sp)
	sw $t3,4($sp)
	#stored 4 registers
	#caught a string
	#preparing to call print
	addiu $sp, $sp, -4
	sw $fp, ($sp)
	addiu $sp, $sp, -4
	sw $ra, ($sp)
	#loading str address to stack
	addiu $sp, $sp, -4
	la $v0, string_label_9
	sw $v0, ($sp)
	move	$fp,$sp
	#jumping to func: print
	jal	print
	#returned from func: print
	#poping 1 params from stack 
	addiu $sp, $sp, 4
	#restoring ra and fp
	lw $ra, ($sp)
	addiu $sp, $sp, 4
	lw $fp, ($sp)
	addiu $sp, $sp, 4
	#restoring 4 previously used registers
	lw $t0,16($sp)
	lw $t1,12($sp)
	lw $t2,8($sp)
	lw $t3,4($sp)
	#Moving funcRes (if exists) to next free register
	#finished calling print
	#end of statement jump
	j	bp_label_15
	#switch init label
bp_label_14:
	beq $t3,4,bp_label_13
bp_label_15:
	#end of statement jump
	j	bp_label_16
bp_label_16:
	#Adding an extre return just in case
	jr	$ra
