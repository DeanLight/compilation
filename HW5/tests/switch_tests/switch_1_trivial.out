Loaded: ./exceptions.s
one
two
three
four